src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -107,21 +111,25 @@ public TemplateCache(JObject? contentJobject, ILogger logger)
         [JsonProperty]
         public Dictionary<string, DateTime> MountPointsInfo { get; }
 
-        private ILocalizationLocator? GetBestLocalizationLocatorMatch(IReadOnlyList<ILocalizationLocator> localizations, string identity)
+        private ILocalizationLocator? GetBestLocalizationLocatorMatch(IScanTemplateInfo template)
         {
-            IEnumerable<ILocalizationLocator> localizationsForTemplate = localizations.Where(locator => locator.Identity.Equals(identity, StringComparison.OrdinalIgnoreCase));
+            if (template.Localizations is null)
+            {
+                return null;
+            }
 
-            if (!localizations.Any())
+            if (!template.Localizations.Any())
             {
                 return null;
             }
-            IEnumerable<string> availableLocalizations = localizationsForTemplate.Select(locator => locator.Locale);
+
+            IEnumerable<string> availableLocalizations = template.Localizations.Keys;
             string? bestMatch = GetBestLocaleMatch(availableLocalizations);
             if (string.IsNullOrWhiteSpace(bestMatch))
             {
                 return null;
             }
-            return localizationsForTemplate.FirstOrDefault(locator => locator.Locale == bestMatch);
+            return template.Localizations[bestMatch!];
```

Comment:
Missing localization will throw here.

Reply:
`GetBestLocaleMatch` guarantee that the key is present.